sklearn/kernel_approximation.py

```diff
@@ -436,6 +436,7 @@ def transform(self, X):
         """
         check_is_fitted(self)
 
+        self._check_feature_names(X, reset=False)
         X = as_float_array(X, copy=True)
         X = self._validate_data(X, copy=False, reset=False)
```

Comment:
Reading the source code of `as_float_array`, I have the feeling that this helper semantics are very obscure and preserved mostly for backward compat.

In this particular case, couldn't we just remove the call to `as_float_array` and let `_validate_data` check the feature names as usual?